pkg/checks/external/nodeCheck/main.go

```diff
@@ -0,0 +1,203 @@
+package nodeCheck
+
+import (
+	"errors"
+	"net"
+	"os"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/pkg/api/v1/pod"
+)
+
+// WaitForKuberhealthyEndpoint first checks the node's age to make sure it's not less than three minutes old. If so, it
+// checks if the kuberhealthy endpoint (KH_REPORTING_URL) can be hit, determining whether or not the proper netfilter
+// rules have been set up on the node in order to hit the kuberhealthy service.
+func WaitForKuberhealthyEndpoint(client *kubernetes.Clientset, namespace string, kuberhealthyEndpoint string) {
+
+	khPod, err := getKHPod(client, namespace)
+	if err != nil {
+		// Just log the error and return since this check will not work if there's an error getting the podName
+		log.Errorln("Error getting kuberhealthy pod:", err)
+		return
+	}
+
+	log.Infoln("Pod is on node:", khPod.Spec.NodeName)
+
+	nodeMinAge := time.Minute * 3
+	nodeNew, err := checkNodeNew(client, khPod, nodeMinAge)
+	if err != nil {
+		// Just log the error and return since this check will not work if there's an error getting the node
+		log.Errorln("Error checking if node is new:", khPod.Spec.NodeName, err)
+		return
+	}
+
+	if nodeNew {
+		log.Infoln("Node is less than", nodeMinAge, "old. Checking if the kuberhealthy endpoint is ready.")
+		select {
+		case <-waitForKuberhealthyEndpointReady(kuberhealthyEndpoint):
+			log.Infoln("Kuberhealthy endpoint:", kuberhealthyEndpoint, "is ready. Proceeding to run check.")
+			return
+		case <-time.After(time.Minute):
+			log.Errorln("Time out checking if kuberhealthy endpoint:", kuberhealthyEndpoint, "is ready. " +
+				"Check may or may not complete successfully.")
+		}
+	}
+}
+
+// WaitForKubeProxy first checks the node's age to make sure it's not less than three minutes old. If so, it sleep for
+// one minute and checks if kube proxy is ready and running on the node before running the check.
+func WaitForKubeProxy(client *kubernetes.Clientset, namespace string) {
+
+	khPod, err := getKHPod(client, namespace)
+	if err != nil {
+		// Just log the error and return since this check will not work if there's an error getting the podName
+		log.Errorln("Error getting kuberhealthy pod:", err)
+		return
+	}
+	log.Infoln("Pod is on node:", khPod.Spec.NodeName)
+
+	nodeMinAge := time.Minute * 3
+	nodeNew, err := checkNodeNew(client, khPod, nodeMinAge)
+	if err != nil {
+		// Just log the error and return since this check will not work if there's an error getting the node
+		log.Errorln("Error checking if node is new:", khPod.Spec.NodeName, err)
+		return
+	}
+
+	if nodeNew {
+		sleepDuration := time.Minute
+		log.Infoln("Node is less than", nodeMinAge, "old. Sleeping for", sleepDuration)
+		time.Sleep(sleepDuration)
+
```

Comment:
not sure if we still want to sleep for a minute before checking kube-proxy. We time out after a minute waiting for kube-proxy to be ready and running.